src/java.base/share/classes/java/lang/foreign/GroupLayout.java

```diff
@@ -82,73 +82,81 @@ public sealed interface GroupLayout extends MemoryLayout permits StructLayout, U
      * of the provided {@link Record} {@code type} by means of matching the names of the
      * record components with the names of the elements in this group layout}
      * <p>
-     * The mapping between the record type and this memory layout is defined as follows;
+     * A mapper {@code M} between a record type {@code R}, a memory layout {@code L} and an {@code offset}
+     * (which is zero for the initial invocation of this method) is defined as follows {@code M(type, layout, offset)};
      * <p>
-     * Let <em>R</em> be the provided record {@code type} with the constituent components
-     * <em>C<sub>0</sub></em>, ..., <em>C<sub>N-1</sub></em>, where <em>N</em> is non-negative.
+     * Let {@code R} comprise its the constituent components
+     * {@code c_1, c_2}, ..., c_N}, where {@code N} is the (non-negative) number of components of {@code R}.
      * <p>
-     * Let <em>L</em> be this group layout with the elements <em>E<sub>0</sub></em>, ..., <em>E<sub>M-1</sub></em>
-     * , where <em>M</em> {@code >=} <em>N</em>.
+     * Let {@code f_a(MemorySegment ms)} be a function that takes
+     * a {@link MemorySegment} {@code ms} and produces a value for {@code c_a}.
      * <p>
-     * Then, for each <em>C<sub>a</sub></em> (<em>a</em> {@code <} <em>N</em>), there must be a corresponding distinct
-     * <em>E<sub>b</sub></em> such that the {@link MemoryLayout#name() name} of <em>C<sub>a</sub></em>
-     * {@link Object#equals(Object) equals} the {@link RecordComponent#getName() name} of the <em>E<sub>b</sub></em> and:
+     * Let {@code L} be a group layout with the elements {@code e_1, e_2, ..., e_M}, where {@code M >= N}.
+     * <p>
+     * Let {@code offsetBetween(GroupLayout layout, MemoryLayout target)} be a function that, via
```

Comment:
I would use two letters as the argument of `offsetBetween`. E.g.
"Let offsetOf(L, LL) be a function that, via ... , can compute the offset from the layout L to a sub-layout LL"
Note the change of name. I don't think "between" is a good choice of word.